shell.nix

```diff
@@ -36,7 +36,7 @@ in
 pkgs.mkShell {
   buildInputs = with pkgs; [
     git
-    go_1_18
+    go_1_21
```

Comment:
This can probably just be `go`; we're much quicker to pick up new versions and I don't think we're very likely to be in a situation where this version is ahead of our support window. Currently, it's on 1.22.7: https://search.nixos.org/packages?channel=unstable&show=go&from=0&size=50&sort=relevance&type=packages&query=go